woocommerce/woocommerce-ios
# Blaze: Add banner on My Store tab

Closes: #10157 

⚠️  Please make sure #10135 is reviewed and merged.


## Description

This PR adds the Blaze banner to the My Store tab. Changes include:
- Updated `StoreOnboardingView` to remove the top and bottom spacer in collapsed mode and added spacing to the `DashboardViewController`'s container stack view instead.
- Since the banner should only be displayed when the store has at least one product, we need to check for products in the store when determining the banner visibility. We have an action in `ProductAction` for that check, but it was named `checkProductsOnboardingEligibility`. To make the action more generic, I renamed it to `checkIfStoreHasProducts` and updated the logic of product onboarding banner accordingly.
- Updated the `DashboardViewModel` to check for the visibility of the Blaze banner and support hiding it when necessary.
- Updated the `DashboardViewController` to:
  - insert the Blaze banner below the store onboarding card or the header view
  - recheck the Blaze banner upon pull-to-refresh.
- Also: updated `ProductsViewController` to hide the Blaze banner if the product list is updated between the empty and non-empty states.

## Testing instructions

- Log in as an admin to a public WPCom site with at least one product to ensure you are eligible to create a Blaze campaign.
- If the site has no products, notice that the product onboarding banner is displayed and the Blaze banner is not. The Blaze banner should not be displayed on the product list screen either.
- Add a product to the store and refresh the dashboard screen, the Blaze banner should be displayed on both the product list screen and the dashboard screen.
- If the store still has incomplete onboarding tasks, the Blaze banner should be displayed below the onboarding card.
- Filter "blaze" in Xcode console, notice that the source for the events on the My Store tab is `my_store`.

## Screenshots

| With JITM banner | With onboarding card | Without other banners |
| ----- | ----- | ----- |
| <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/c1872b2e-7e2d-4224-a5d8-d357d7baf799" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/59d6439a-a895-4c06-b114-b552b514fc5f" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/0c858edf-491b-40a6-837e-f365da194f3f" width=320 /> |


---
